gfx.c

```diff
@@ -19,37 +19,37 @@
 
 #define DOWNCONVERT_BIT_DEPTH(x) ((x) / 8)
 
-static void AdvanceMetatilePosition(int *subTileX, int *subTileY, int *metatileX, int *metatileY, int metatilesWide, int metatileWidth, int metatileHeight)
+static void AdvanceTilePosition(int *tilesSoFar, int *rowsSoFar, int *chunkStartX, int *chunkStartY, int chunksWide, int tilesPerRow, int rowsPerChunk)
 {
-    (*subTileX)++;
-    if (*subTileX == metatileWidth) {
-        *subTileX = 0;
-        (*subTileY)++;
-        if (*subTileY == metatileHeight) {
-            *subTileY = 0;
-            (*metatileX)++;
-            if (*metatileX == metatilesWide) {
-                *metatileX = 0;
-                (*metatileY)++;
+    (*tilesSoFar)++;
+    if (*tilesSoFar == tilesPerRow) {
+        *tilesSoFar = 0;
+        (*rowsSoFar)++;
+        if (*rowsSoFar == rowsPerChunk) {
+            *rowsSoFar = 0;
+            (*chunkStartX)++;
+            if (*chunkStartX == chunksWide) {
+                *chunkStartX = 0;
+                (*chunkStartY)++;
             }
         }
     }
 }
 
-static void ConvertFromTiles1Bpp(unsigned char *src, unsigned char *dest, int numTiles, int metatilesWide, int metatileWidth, int metatileHeight, bool invertColors)
+static void ConvertFromTiles1Bpp(unsigned char *src, unsigned char *dest, int numTiles, int chunksWide, int tilesPerRow, int rowsPerChunk, bool invertColors)
 {
-    int subTileX = 0;
-    int subTileY = 0;
-    int metatileX = 0;
-    int metatileY = 0;
-    int pitch = metatilesWide * metatileWidth;
+    int tilesSoFar = 0;
+    int rowsSoFar = 0;
+    int chunkStartX = 0;
+    int chunkStartY = 0;
+    int pitch = chunksWide * tilesPerRow;
 
     for (int i = 0; i < numTiles; i++) {
         for (int j = 0; j < 8; j++) {
-            int destY = (metatileY * metatileHeight + subTileY) * 8 + j;
-            int destX = metatileX * metatileWidth + subTileX;
+            int idxComponent1 = (chunkStartY * rowsPerChunk + rowsSoFar) * 8 + j;
```

Comment:
idxComponentY would be better than idxComponent1, same for x

Reply:
ok I went in and made the proposed changes.